nifi-commons/nifi-flow-encryptor/src/main/java/org/apache/nifi/flow/encryptor/XmlFlowEncryptor.java

```diff
@@ -0,0 +1,120 @@
+                if (event.getEventType() == XMLEvent.START_DOCUMENT) {
+                    final StartDocument startDocument = (StartDocument) event;
+                    handleXmlStartDocument(startDocument, outputStream);
+                } else if (event.getEventType() == XMLEvent.CHARACTERS) {
+                    final Characters characters = event.asCharacters();
+                    final String value = characters.getData();
+                    final Matcher matcher = ENCRYPTED_PATTERN.matcher(value);
+                    if (matcher.matches()) {
+                        final String processedValue = getOutputEncrypted(matcher.group(FIRST_GROUP), inputEncryptor, outputEncryptor);
+                        writer.add(eventFactory.createCharacters(processedValue));
+                    } else {
+                        writer.add(characters);
+                    }
+                } else if (event.getEventType() == XMLEvent.COMMENT) {
+                    writer.add(event);
+                    final Characters newLine = eventFactory.createCharacters("\n");
+                    writer.add(newLine);
+                } else if (event.getEventType() == XMLEvent.END_DOCUMENT) {
+                    final Characters newLine = eventFactory.createCharacters("\n");
+                    writer.add(newLine);
+                } else {
+                    writer.add(event);
+                }
+            }
+            writer.flush();
+            writer.close();
+            reader.close();
+            outputStream.close();
+            inputStream.close();
+        } catch (XMLStreamException e) {
+            logger.error("Failed Processing Flow Configuration: XMLStreamException occurred");
+            e.printStackTrace();
+        } catch (IOException e) {
+            throw new UncheckedIOException("Failed Processing Flow Configuration", e);
+        }
+    }
+
+    private String getOutputEncrypted(final String inputEncrypted, final PropertyEncryptor inputEncryptor, final PropertyEncryptor outputEncryptor) {
+        final String inputDecrypted = inputEncryptor.decrypt(inputEncrypted);
+        final String outputEncrypted = outputEncryptor.encrypt(inputDecrypted);
+        return String.format(ENCRYPTED_FORMAT, outputEncrypted);
+    }
+
+    private void handleXmlStartDocument(final StartDocument startDocument, final OutputStream outputStream) {
+        final PrintWriter writer = new PrintWriter(new OutputStreamWriter(outputStream));
+        final String version = startDocument.getVersion();
+        final String encoding = startDocument.getCharacterEncodingScheme();
+        String standalone = "";
+        if (startDocument.standaloneSet()) {
+            standalone = startDocument.isStandalone() ? " standalone=\"yes\"" : " standalone=\"no\"";
+        }
+        final String line = String.format("<?xml version=\"%s\" encoding=\"%s\"%s?>", version, encoding, standalone);
+        writer.println(line);
+        writer.flush();
+    }
```

Comment:
Does XMLEventWriter not handle the `standalone` property on the `StartDocument` event? The default value is `no`, which is what all flow.xml documents should have. It seems better to use just use the XMLEventWriter behavior instead of writing a custom string.

Reply:
The XMLEventWriter does not handle the standalone property whether its set to `yes` or `no`, which is why I did this

Reply:
Thanks for the explanation. Since `standalone="no"` is the default value when not specified, and that is always the preferred setting for the flow configuration, it seems better to just remove the custom handling in this method.